docs/changelog.txt

```diff
@@ -54,22 +54,42 @@ Template for new versions:
 ## New Tools
 
 ## New Features
+- ``cuboid``: construct from ``df::map_block*``, ``forBlock`` iterator to access map blocks in cuboid
+- ``cuboid``: ``clamp(cuboid other)``, ``clampNew(cuboid other)`` for cuboid intersection. ``clampNew`` returns new cuboid instead of modifying.
+- ``Units::isUnitInBox``, ``getUnitsInBox``: add versions that take cuboid, filter fn parameter for ``getUnitsInBox``
+- ``Units::getProfession``: fn to return profession type accounding for fake identity
 
 ## Fixes
 - `changelayer`: fix incorrect lookup of geological region in multi-region embarks
 - Fix handling of multi-line text when interacting with the system clipboard on Windows
+- Fix ``cuboid`` ``df::coord`` constructors not initializing properly
+- ``Units::isVisible``: account for units in cages
+- ``Units::isHidingCurse``: don't check ``hunt_target``, which is obsolete
+- ``Units::getReadableName``: fix position of "Ghostly", use ``curse.name`` instead of iterating syndrome name effects
 
 ## Misc Improvements
 - `tweak`: improve performance of ``adamantine-cloth-wear`` tweak
+- ``setAreaAquifer``, ``removeAreaAquifer``: make use of cuboid iters to simplify code
+- ``Units::isNaked``: now only checks equipped items (including rings, for now). Setting bool ``no_items`` to true checks empty inventory like before.
+- ``Units::isUndead``: bool ``include_vamps`` renamed to ``hiding_curse``. Fn now checks that instead of bloodsucker syndrome.
+- ``Units::isDanger``: added bool ``hiding_curse``, passed to ``isUndead`` to avoid spoilers
+- ``Units::getRaceChildName``, ``getRaceChildNameById``, ``getRaceBabyName``, ``getRaceBabyNameById``: bool ``plural`` to get plural form
+- ``Units::getProfessionName``: bool ``land_title`` to append "of Sitename" where applicable, use Prisoner/Slave and noble spouse titles (controlled by ``ignore_noble``)
+- `plant`: detect trees via branches/leaves for cuboid selection
 
 ## Documentation
+- ``Units.h``: more comments on what each fn does
+- Improve ``Units`` module Lua API docs, including some fns that were completely unlisted
 
 ## API
+- ``Units::getRaceReadableName``, ``getRaceReadableNameById``, ``getRaceNamePluralById``: add missing fns to Lua API
 
 ## Lua
 - ``ZScreen``: new ``defocused`` property for starting screens without keyboard focus
 
 ## Removed
+- ``cuboid::clamp(bool block)``: renamed to ``cuboid::clampMap(bool block)``, name taken by ``cuboid::clamp(cuboid other)``
+- ``Units::MAX_COLORS``, ``Units::findIndexById``, ``Units::getNumUnits``, ``Units::getUnit``: not used anywhere, covered by structure defs and vector size
```

Comment:
looks like stonesense needs to be modified to not use MAX_COLORS:
```
/home/runner/work/dfhack/dfhack/plugins/stonesense/commonTypes.h:306:35: error: ‘MAX_COLORS’ is not a member of ‘DFHack::Units’
```

Reply:
Missed that since I don't build stonesense. What does this value actually represent? My best guess is:
```
constexpr int32_t MAX_COLORS = (ENUM_LAST_ITEM(curses_color) + 1) * 2 - 1; // Or just DFHack::COLOR_MAX
```
but that's the index of the last color (`COLOR_WHITE'), not the number of elements.

Stonesense uses it like this:
commonTypes.h
```
uint32_t color[DFHack::Units::MAX_COLORS];
```
Creatures.cpp
```
furball.nbcolors = source->appearance.colors.size();
if (furball.nbcolors > DFHack::Units::MAX_COLORS)
    furball.nbcolors = DFHack::Units::MAX_COLORS;
```
which implies it's a number of elements, and something to do with `unit.appearance.colors` (which I've no info on).

Reply:
This is some pretty terrible code:
https://github.com/DFHack/stonesense/blob/6dc778366b851634ad2ff43948b987aa98c3104b/GUI.cpp#L684
I had a mini panic attack deciphering it.

Essentially:
```
// Stuff from struct SS_Unit
uint32_t nbcolors = std::min((int32_t)unit->appearance.colors.size(), Units::MAX_COLORS);
uint32_t color[Units::MAX_COLORS];

for(uint32_t i = 0; i < nbcolors; i++) {
    color[i] = unit->appearance.colors[i];
}
// End of SS_Unit stuff

auto cr = df::creature_raw::find(unit->race)->caste.at(unit->caste);
for (unsigned int j = 0; j < nbcolors; j++) {
    int32_t cr_color = cr->color_modifiers[j]->pattern_index[color[j]];
    auto desc_pat = world->raws.descriptors.patterns[cr_color];
    for (auto color_idx : desc_pat->colors) {
        auto desc_color = world->raws.descriptors.colors[color_idx];
        // Draw a rectangle using (desc_color->red, desc_color->green, desc_color->blue) floats
    }
}
```
In other words, `Units::MAX_COLORS` is being used completely incorrectly and there's zero reason to be using it.

Easiest fix would probably be to convert `SS_Unit.color` into a `std::vector` and set `SS_Unit.nbcolor` to `SS_Unit.color.size()`.

Reply:
There's a reason stonesense is so crashy : (